### PR TITLE
[RNMobile][Monorepo] Update react-native-* packages from gutenberg-mobile

### PR DESCRIPTION
**Pull request by Tug**
## Description
This PR's goal is to bring the latest changes from gutenberg-mobile to the monorepo branch.
Thus, it also gets gutenberg `master` recent changes in are the 2 repo needs to be in sync.

Most commits are from the gutenberg-mobile repo, the ones that are not are merge commits (from gb-mobile develop or gb master)

> Note: We have a large amount of changes in gutenberg-mobile due to the recent react-native upgrade.

## Testing Instructions
Try running the following commands:
- `npm run native test`
- `npm run test-unit:native`
- `npm run native start:reset` then `curl --silent --output /dev/null "http://localhost:8081/index.bundle?platform=android&dev=true&minify=false"` to generate a bundle for android
- Check CI jobs on this PR, only the "License compatibility" one should be failing (this is still to be addressed)

## Types of changes
Port of commits

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-editor/metro.config.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+		getTransformOptions: async () => ( {
+			transform: {
+				experimentalImportSupport: false,
+				inlineRequires: false,
+			},
+		} ),
+	},
+};
+
+console.log( 'metro config ', module.exports );
```
Comment (dratwas) [inline packages/react-native-editor/metro.config.js]: Do you want to leave it here or is it temporary?

**Comment by dratwas** [comment]
>npm run start:reset

I believe that this should be `npm run native start:reset`
With that command, everything seems to work.

Also, I'm not sure if we should leave the console log with the whole metro config at the start. Overall it looks good :)

**Comment by gziolo** [comment]
@Tug and @dratwas – do you want me to validate anything specific?

File: packages/react-native-editor/metro.config.js (hunk 1 above)
Comment (Tug) [reply packages/react-native-editor/metro.config.js]: Good point, this should be removed

**Comment by Tug** [comment]
@dratwas thanks for reviewing! I might actually update once more since we're at it (also gutenberg-mobile changes are out of sync with gutenberg here and it causes problem when we try to run the app)

@gziolo Just wanted you to know about it :)
It's a lot but it shouldn't be modifying any web code, most of the diff is simply bringing the monorepo branch up to date with `master` (we can't easily rebase given the amount of commits this branch has and the never ending rebase hell it would cause)

**Comment by Tug** [comment]
@gziolo Actually we'll need an admin to merge without squashing. Let's ping once it's ready

**Comment by Tug** [comment]
@dratwas it's ready for another pass :)

Btw, you can try to actually build the dev bundle by requesting:
http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false
http://localhost:8081/index.bundle?platform=android&dev=true&minify=false

**Review by ceyhun — APPROVED** [review]
I was able to successfully run all the tests 👍

**Comment by gziolo** [comment]
> @gziolo Actually we'll need an admin to merge without squashing. Let's ping once it's ready

It's disabled for me at the moment. @youknowriad has superpowers to change that :)

**Comment by Tug** [comment]
@hypest I think you have the superpowers as well, could you try to do it?

For reference, the reason we need this is because until we haven’t made the switch to use `gutenberg` as our main repo, we’ll want to pull changes from `gutenberg-mobile` with `git subtree pull` and that won’t work if we rewrite history!

**Comment by hypest** [comment]
Merging with "Merge commit" option set to true 👍

**Pull request merged**